components/windows-service/HabService.cs

```diff
@@ -108,6 +108,17 @@ protected override void OnStart(string[] args)
             }
         }
 
+        private static void ConfigureEnvironment()
+        {
+            var envPrefix = "ENV_";
+            foreach(var key in ConfigurationManager.AppSettings.AllKeys) {
+                var i = key.Trim().ToUpper();
```

Comment:
Curious about when whitespace can appear (and Trim is needed). Is there always whitespace, or is that only when someone is doing something wrong? The effect here is to fold "ENV_WIERD " (with space) and "ENV_WIERD" (without space) together; which takes effect depends on the order from AllKeys. 

There's a similar effect from the ToUpper; if both "env_case" and "ENV_CASE" are present only one will passed.

Would it make sense to be more restrictive, and only pass 'well formed' ENV_* variables (no whitespace, all upper, etc.)

It's been a while since I've done C#/windows, so very well could be missing something.

Reply:
The `trim` is really defensive handling of a config file content like:

```
<add key="   HAB_FEAT_IGNORE_SIGNALS   " value="true" />
```

If some one had multiple entries for the same key. For example one with spaces and the other without, it seems like it would be unintentional. However even if we omitted the trim, windows  would effectively trim the key when we added it to the environment.

Same holds true for the `ToUpper`. On Windows, environment var keys are case insensitive. So it would likely be unintentional to add 2 keys that only vary in casing.

The main benefit of trimmimg and upper casing the key is it just makes it simpler to do the `StartsWith("ENV_")` check.

Reply:
Thanks, makes sense to me.